tvb_storage/tvb/storage/storage_interface.py

```diff
@@ -0,0 +1,469 @@
+        self.data_encryption_handler.set_project_inactive(project)
+
+    def push_folder_to_sync(self, project_name):
+        project_folder = self.get_project_folder(project_name)
+        self.data_encryption_handler.push_folder_to_sync(project_folder)
+
+    @staticmethod
+    def encryption_enabled():
+        return DataEncryptionHandler.encryption_enabled()
+
+    @staticmethod
+    def startup_cleanup():
+        return DataEncryptionHandler.startup_cleanup()
+
+    @staticmethod
+    def project_key_path(project_name):
+        return DataEncryptionHandler.project_key_path(project_name)
+
+    # Folders Queue Consumer methods start here #
+
+    def run(self):
+        return self.folders_queue_consumer.run()
+
+    def start(self):
+        self.folders_queue_consumer.start()
+
+    def mark_stop(self):
+        self.folders_queue_consumer.mark_stop()
+
+    def join(self):
+        self.folders_queue_consumer.join()
+
+    # Generic methods start here
+
+    def ends_with_tvb_file_extension(self, file):
+        return file.endswith(self.TVB_FILE_EXTENSION)
+
+    def ends_with_tvb_storage_file_extension(self, file):
+        return file.endswith(self.TVB_STORAGE_FILE_EXTENSION)
+
+    def rename_project(self, current_proj_name, new_name):
+        project_folder = self.get_project_folder(current_proj_name)
+        if self.encryption_enabled() and not self.is_in_usage(project_folder):
+            raise RenameWhileSyncEncryptingException(
+                "A project can not be renamed while sync encryption operations are running")
+        self.rename_project_structure(current_proj_name, new_name)
+        encrypted_path = self.compute_encrypted_folder_path(project_folder)
+        if os.path.exists(encrypted_path):
+            new_encrypted_path = self.compute_encrypted_folder_path(
+                self.get_project_folder(new_name))
+            os.rename(encrypted_path, new_encrypted_path)
+
+    def remove_project(self, project):
+        project_folder = self.get_project_folder(project.name)
+        self.remove_project_structure(project.name)
+        encrypted_path = self.compute_encrypted_folder_path(project_folder)
+        if os.path.exists(encrypted_path):
+            self.remove_folder(encrypted_path)
+        if os.path.exists(self.project_key_path(project.id)):
+            os.remove(self.project_key_path(project.id))
+
+    def move_datatype_with_sync(self, to_project, to_project_path, new_op_id, full_path, vm_full_path):
+        self.set_project_active(to_project)
+        self.sync_folders(to_project_path)
+
+        self.move_datatype(to_project.name, str(new_op_id), full_path)
+        self.move_datatype(to_project.name, str(new_op_id), vm_full_path)
+
+        self.sync_folders(to_project_path)
+        self.set_project_inactive(to_project)
+
+    def export_datatypes(self, paths, operation):
+        op_folder = self.get_operation_folder(operation.project.name, operation.id)
```

Comment:
Not used directly by framework, could move deeper in tvb_storage module

Reply:
Maybe a new file for export related methods?